src/glarea.rs

```diff
@@ -0,0 +1,203 @@
+        unsafe {
+            let program = Gl.CreateProgram();
+            Gl.AttachShader(program, vs);
+            Gl.AttachShader(program, fs);
+            Gl.LinkProgram(program);
+
+            // Get the link status
+            let mut status = epoxy::FALSE as GLint;
+            Gl.GetProgramiv(program, epoxy::LINK_STATUS, &mut status);
+
+            // Fail on error
+            if status != (epoxy::TRUE as GLint) {
+                let mut len: GLint = 0;
+                Gl.GetProgramiv(program, epoxy::INFO_LOG_LENGTH, &mut len);
+                let mut buf = vec![0u8; len as usize - 1];
+                Gl.GetProgramInfoLog(program, len, ptr::null_mut(), buf.as_mut_ptr() as *mut GLchar);
+                panic!("Error linking shader");
+            }
+
+            program
+        }
+    }
+
+    pub fn main() {
+        if gtk::init().is_err() {
+            println!("Failed to initialize GTK.");
+            return;
+        }
+
+        let window = Window::new(gtk::WindowType::Toplevel).unwrap();
+
+        let glarea = GLArea::new().unwrap();
+
+        window.connect_delete_event(|_, _| {
+            gtk::main_quit();
+            Inhibit(false)
+        });
+
+        glarea.connect_realize(clone!(glarea; |_widget| {
+            glarea.make_current();
+
+            let vertices: [GLfloat; 15] = [
+                0.0, 0.5, 1.0, 0.0, 0.0,
+                0.5, -0.5, 0.0, 1.0, 0.0,
+                -0.5, -0.5, 0.0, 0.0, 1.0,
+            ];
+
+            let vert_shader_src = r#"
+                #version 140
+
+                in vec2 position;
+                in vec3 color;
+
+                out vec3 vertex_color;
+
+                void main() {
+                    vertex_color = color;
+                    gl_Position = vec4(position, 0.0, 1.0);
+                }"#;
+
+            let frag_shader_src = r#"
+                #version 140
+
+                in vec3 vertex_color;
+
+                out vec4 color;
+
+                void main() {
+                    color = vec4(vertex_color, 1.0);
+                }"#;
+
+            let vs = compile_shader(vert_shader_src, epoxy::VERTEX_SHADER);
+            let fs = compile_shader(frag_shader_src, epoxy::FRAGMENT_SHADER);
+            let program = link_program(vs, fs);
+
+            let mut vao: GLuint = 0;
+            let mut vbo: GLuint = 0;
+
+            unsafe {
+                Gl.GenVertexArrays(1, &mut vao);
+                Gl.BindVertexArray(vao);
+
+                Gl.GenBuffers(1, &mut vbo);
+                Gl.BindBuffer(epoxy::ARRAY_BUFFER, vbo);
+                Gl.BufferData(epoxy::ARRAY_BUFFER,
+                              (vertices.len() * mem::size_of::<GLfloat>()) as GLsizeiptr,
+                              mem::transmute(&vertices[0]),
+                              epoxy::STATIC_DRAW);
+
+                Gl.UseProgram(program);
+                Gl.BindFragDataLocation(program, 0, ffi::CString::new("color").unwrap().as_ptr());
```

Comment:
Would it make sense to define 
```
const COLOR: &'static u8 = b"color\0";
```
and use `COLOR.as_ptr() as *const i8` in place of `CString` here?.